### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by Cadair**
The objective of this PR is to cleanup the test suite so all test setup is done inside pytest fixtures.

### CODE REVIEW

File: ndcube/tests/conftest.py [hunk 1]
```diff
@@ -13,36 +13,39 @@
...
+        'CDELT4': 5,
+        'CRPIX4': 5,
+        'CRVAL4': 0,
 
-        'CTYPE1': 'TIME    ',
-        'CUNIT1': 'min',
-        'CDELT1': 0.4,
-        'CRPIX1': 0,
-        'CRVAL1': 0,
+        'DATEREF': "2020-01-01T00:00:00"
```
Comment (DanRyanIrish) [inline ndcube/tests/conftest.py:42]: In theory could this also be `'DATE-OBS`?  Or is `'DATEREF` the key that astropy WCS recognises?

File: ndcube/tests/test_ndcube.py [hunk 2]
```diff
@@ -1,819 +1,277 @@
...
+                             ("ndcube_3d_ln_lt_l", np.s_[..., 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[1:2, 1:2, 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[..., 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[:, :, 0]),
+                             ("ndcube_3d_ln_lt_l", np.s_[1:2, 1:2, 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[:, :, 0, 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[..., 0, 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[1:2, 1:2, 1, 1]),
+                         ),
+                         indirect=("ndc",))
```
Comment (DanRyanIrish) [inline ndcube/tests/test_ndcube.py:34]: How does this `indirect` thing work?

File: ndcube/tests/test_ndcube.py [hunk 3]
```diff
@@ -1,819 +1,277 @@
...
+    assert isinstance(sndc.wcs, BaseHighLevelWCS)
+    assert isinstance(sndc.wcs.low_level_wcs, SlicedLowLevelWCS)
+    wcs = sndc.wcs
+    assert wcs.pixel_n_dim == 2
+    assert wcs.world_n_dim == 3
+    assert np.allclose(wcs.array_shape, sndc.data.shape)
+    assert set(wcs.world_axis_physical_types) == {"custom:pos.helioprojective.lat",
+                                                  "custom:pos.helioprojective.lon",
+                                                  "em.wl"}
+    assert np.allclose(wcs.axis_correlation_matrix, np.array([[True, False],
```
Comment (DanRyanIrish) [inline ndcube/tests/test_ndcube.py:126]: Sure this should be a hard assert, rather than `allclose`?

**Review by DanRyanIrish — APPROVED** [review]

File: ndcube/tests/test_ndcube.py (hunk 2 above)
Comment (Cadair) [reply ndcube/tests/test_ndcube.py:34]: It is a froody fixture lookup thing. Basically it lets you lookup a fixture by name. It passes the name specified in the parametrize to the `ndc` fixture which returns the fixture with the given name.

File: ndcube/tests/test_ndcube.py (hunk 3 above)
Comment (Cadair) [reply ndcube/tests/test_ndcube.py:126]: Don't think it matters much.

File: ndcube/tests/conftest.py (hunk 1 above)
Comment (Cadair) [reply ndcube/tests/conftest.py:42]: I think dateref is something slightly different. I think this is a hold over from a SpectralCoord bug.

**Pull request merged**